prov/gni/src/gnix_cq.c

```diff
@@ -588,13 +590,22 @@ DIRECT_FN STATIC ssize_t gnix_cq_readfrom(struct fid_cq *cq, void *buf,
 	return __gnix_cq_sreadfrom(0, cq, buf, count, src_addr, NULL, 0);
 }
 
+/*
+ * TODO: Remove ifdef's below and update api_version checking once upstream
+ * #2772 is resolved.
+ */
```

Comment:
Please remove everything related to `struct util_fabric` below.  It is not even close to right, so I don't want it in the repo, even as a comment.